### DISCUSSION TITLE
about gst-launch jpeg decode

### DISCUSSION THREAD

**Discussion by Ncerzzk** [svpcom/wfb-ng#258] [Questions]
Hi, I want to implement ESP-wifibroadcast, which is using esp32 as the airstation.(with ov2640, jpeg format out)
For now, I have successfully sending out the FEC packets in esp32.
A PC with usb wifi-netcard executing `./wfb_rx -K gs.key -k 4 -n 6 -p 0 wlan1` could recv and decrypt the packets.

While when I use `gst-launch-1.0 udpsrc port=5600 ! jpegdec ! imagefreeze ! glimagesink` to running gst-launch, it shows that the image decoded is uncorrect

because of I'm not familar with gst-launch, so I'd like to ask that is the command correct to decode jpeg stream? so that I could dig to find where the problem is.

**Comment by stevejohn163** [comment]
Can you explain more as to how you got it working?

**Reply by svpcom** [reply]
It depends how you create a jpeg stream. It is rtp based or every udp packet have a full jpeg inside?

**Reply by stevejohn163** [reply]
It is based on sending udp packets with jpeg inside
I'm trying to implement this project where I have an esp32 cam and want to use esp32 as my air unit and then use wfb-ng on the ground station 
[project](https://github.com/RomanLut/hx-esp32-cam-fpv/tree/master)
How exactly can I do it, is there a better way to do it?
Thank you
